core/src/main/scala/chisel3/experimental/Defaulting.scala

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.experimental
+
+import chisel3.{chiselTypeOf, Record, Data, Element}
+
+import scala.collection.immutable.SeqMap
+
+/** Data type for representing any type who has a default value
+  *
+  * Uses the underlying opaque type support.
+  *
+  * @note This API is experimental and subject to change
+  */
+final class Defaulting[T <: Data] private (private[chisel3] val tpe: T, val defaultValue: T) extends Record {
+  requireIsChiselType(tpe, s"Chisel hardware type $tpe must be a pure type, not bound to hardware.")
+  requireIsHardware(defaultValue, s"Default value $defaultValue must be bound to a hardware component")
+
+  /** The underlying hardware component, is either the Chisel data type or hardware component */
```

Comment:
isn't this comment wrong? Isn't the default value the underlying hardware component?

Reply:
I don't think so? The underlying hardware component, (e.g. a field in a bundle) is the `underlying`, and the defined default value is `default`. I'm not sure if I'm misunderstanding your question.. I clarified in the next PR.